### ISSUE TITLE
TypeError: Cannot read property 'action' of undefined

### ISSUE THREAD

**Issue by lucasecdb** [zalmoxisus/redux-devtools-extension#312]
Hi there,

I've received the following error while using the redux devtools on chrome:

![redux devtools bug](https://cloud.githubusercontent.com/assets/10223856/23314785/1f00e714-faa2-11e6-9617-6ceb304cd7a8.PNG)

I've searched the source and found that the error occur on [this file](https://github.com/zalmoxisus/redux-devtools-extension/blob/master/src/app/api/filters.js#L63). I can tell more about the redux "stack" that I'm using, if that's necessary.

I'm using Google Chrome v56.0.2924.87 (64-bit).

**Comment by lucasecdb** [comment]
The redux devtools version is 2.14.0.

**Comment by zalmoxisus** [comment]
If that's the line throwing, it means you're having `actionsBlacklist`, `actionsWhiteleast` or `predicate` option specified right? I just tried with those and cannot replicate.

Could you please share a repo? If not at least the store config.

**Comment by lucasecdb** [comment]
I can't share the repo, unfortunately. By 'store config' you mean the redux state?

I didn't figure out what the problem was, but I reinstalled the extension and the error stopped appearing.

**Comment by zalmoxisus** [comment]
By store config, I mean how you [apply the extension's enhancer](https://github.com/zalmoxisus/redux-devtools-extension#11-basic-store).

As I see from the stack trace, it comes from inside `filterState`, [which was changed in `2.14.1`](https://github.com/zalmoxisus/redux-devtools-extension/commit/293ca862dff977e3865ffccf7641c0d09d536151). Hope that was the fix for this issue.

Please let me know if it happens again.

**Comment by lucasecdb** [comment]
And it's back.

**Comment by lucasecdb** [comment]
The extension enhancer is being applied by composing the enhancers, but it's done a bit differently from what's in the docs.

```javascript
const devtools = window.devToolsExtension || (() => noop => noop);

const middlewares = [ /* middlewares... */ ];

const enhancers = [ applyMiddleware(...middlewares) ];

if (!isProduction) {
  enhancers.push(devtools());
}

const store = createStore(
  createReducer(),
  initialState,
  compose(...enhancers),
);
```

This is how I apply the enhancer, but I tried using the composedEnhancer (with ```window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__```) and the error still occur.

**Comment by zalmoxisus** [comment]
@lucasecdb, the store config looks good. There could be something in `createReducer` or in the meiddlewares preventing from adding an action to the history. The only case I know can cause this is when the extension's enhancer  isn't the last in the compose, but as I see it's not the case here.

We could just skip those actions, but if you can provide a repro to get the issue it would be much better. Maybe you can fork this repo and modify [one of our examples](https://github.com/zalmoxisus/redux-devtools-extension/tree/master/examples) to reproduce the issue.

**Comment by zalmoxisus** [comment]
Should be fixed in `2.14.2`.
